contracts/NodeOperatorRegistry.sol

```diff
@@ -0,0 +1,642 @@
+            "The operator status is not STAKED"
+        );
+        IValidator(no.validatorContract).topUpForFee(msg.sender, _heimdallFee);
+
+        emit TopUpHeimdallFees(id, _heimdallFee);
+    }
+
+    /// @notice Allows to unstake staked tokens after withdraw delay.
+    function unstakeClaim() external override whenNotPaused {
+        uint256 validatorId = operatorOwners[msg.sender];
+        require(validatorId != 0, "Operator not exists");
+
+        Operator.NodeOperator storage no = operators[validatorId];
+        require(
+            no.status == Operator.NodeOperatorStatus.UNSTAKED,
+            "Operator status not UNSTAKED"
+        );
+
+        uint256 amount = IValidator(no.validatorContract).unstakeClaim(
+            msg.sender,
+            validatorId
+        );
+
+        no.status = Operator.NodeOperatorStatus.EXIT;
+        state.totalUnstakedNodeOpearator--;
+        state.totalExitNodeOpearator++;
+
+        emit ClaimUnstake(validatorId, msg.sender, amount);
+    }
+
+    /// @notice Get validator id by user address.
+    /// @param _validatorId validatorId.
+    /// @return Returns the validator total staked.
+    function validatorStake(uint256 _validatorId)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        return IStakeManager(state.stakeManager).validatorStake(_validatorId);
+    }
+
+    /// @notice Get validator total stake.
+    /// @param _user user address.
+    /// @return Returns the validatorId of an address.
+    function getValidatorId(address _user)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        return IStakeManager(state.stakeManager).getValidatorId(_user);
+    }
+
+    /// @notice Get validatorShare contract address.
+    /// @dev Get validatorShare contract address.
+    /// @param _validatorId Validator Id
+    /// @return Returns the address of the validatorShare contract.
+    function getValidatorContract(uint256 _validatorId)
+        external
+        view
+        override
+        returns (address)
+    {
+        return
+            IStakeManager(state.stakeManager).getValidatorContract(
+                _validatorId
+            );
+    }
+
+    /// @notice Allows to withdraw rewards from the validator.
+    /// @dev Allows to withdraw rewards from the validator using the _validatorId. Only the
+    /// owner can request withdraw in this the owner is this contract. This  functions is called
+    /// by a lido contract.
+    function withdrawRewards()
+        external
+        override
+        whenNotPaused
+        returns (uint256[] memory, address[] memory)
+    {
+        require(msg.sender == state.lido, "Caller is not the lido contract");
+        uint256[] memory shares = new uint256[](state.totalStakedNodeOpearator);
+        address[] memory recipient = new address[](
+            state.totalStakedNodeOpearator
+        );
+        uint256 index = 0;
```

Comment:
Maybe consider renaming this index variable because you already have iterator called idx inside for loop. Just to improve readibilidy.